### DISCUSSION TITLE
brew install ffmpeg error, can you correct me, thank you!

### DISCUSSION THREAD

**Discussion by ezewu** [Homebrew/discussions#1765] [Everyday usage]
```
==> Installing ffmpeg dependency: rust
==> ./configure --prefix=/usr/local/Cellar/rust/1.53.0 --release-channel=stable
==> make
Last 15 lines from /Users/wu/Library/Logs/Homebrew/rust/02.make:
using ::isunordered;
      ~~^
13 errors generated.
[10/2716] Building CXX object lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangle.cpp.o
[11/2716] Building CXX object lib/Demangle/CMakeFiles/LLVMDemangle.dir/ItaniumDemangle.cpp.o
ninja: build stopped: subcommand failed.
thread 'main' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', /Users/wu/Library/Caches/Homebrew/cargo_cache/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.44/src/lib.rs:885:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
	finished in 38.541 seconds
failed to run: /private/tmp/rust-20210702-23625-j3l8tp/rustc-1.53.0-src/build/bootstrap/debug/bootstrap build --stage 2
Build completed unsuccessfully in 0:03:24
make: *** [all] Error 1

Do not report this issue to Homebrew/brew or Homebrew/core!

These open issues may also help:
rust: use system curl only on macOS https://github.com/Homebrew/homebrew-core/pull/79727
Rust-dependent formulae on Apple Silicon - upstream issue tracker https://github.com/Homebrew/homebrew-core/issues/68301

Error: You are using macOS 10.13.
We (and Apple) do not provide support for this old version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
old version.
```

**Comment by SMillerDev** [comment]
This is the key issue in the output:

> Error: You are using macOS 10.13.
We (and Apple) do not provide support for this old version.
You will encounter build failures with some formulae.

**Reply by ezewu** [reply]
Now he is outputting this

Error: rust 1.53.0 did not build
Logs:
     /Users/wu/Library/Logs/Homebrew/rust/00.options.out
     /Users/wu/Library/Logs/Homebrew/rust/02.make.cc
     /Users/wu/Library/Logs/Homebrew/rust/01.configure
     /Users/wu/Library/Logs/Homebrew/rust/build
     /Users/wu/Library/Logs/Homebrew/rust/02.make
     /Users/wu/Library/Logs/Homebrew/rust/vendor
Do not report this issue to Homebrew/brew or Homebrew/core!

These open issues may also help:
rust: use system curl only on macOS https://github.com/Homebrew/homebrew-core/pull/79727
Rust-dependent formulae on Apple Silicon - upstream issue tracker https://github.com/Homebrew/homebrew-core/issues/68301

Error: You are using macOS 10.13.
We (and Apple) do not provide support for this old version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
old version.

**Reply by gromgit** [reply]
As @SMillerDev pointed out:
```
Error: You are using macOS 10.13.
We (and Apple) do not provide support for this old version.
You will encounter build failures with some formulae.
```
In short, High Sierra is no longer supported, so you have to fix any build errors yourself. You can start by looking at the log files mentioned:
```
/Users/wu/Library/Logs/Homebrew/rust/00.options.out
/Users/wu/Library/Logs/Homebrew/rust/02.make.cc
/Users/wu/Library/Logs/Homebrew/rust/01.configure
/Users/wu/Library/Logs/Homebrew/rust/build
/Users/wu/Library/Logs/Homebrew/rust/02.make
/Users/wu/Library/Logs/Homebrew/rust/vendor
```

**Reply by ezewu** [reply]
OK, thanks

**Reply by usoppp16** [reply]
did it work out for you because i have the same exact problem

**Comment by thaotx-zigvy** [comment]
Im facing same issue. I installed "rustup" and use it to install rust. But homebrew still install it?

**Reply by SMillerDev** [reply]
That's an entirely different issue. But brew will always install any dependencies it needs, if you decide to install something through another means brew will happily install it's own version next to it.

**Reply by tjluoma** [reply]
I recently had a working High Sierra install with `ffmpeg` which seemed to break after running `brew update`. Which was frustrating.
